core/lsp/Task.h

```diff
@@ -0,0 +1,17 @@
+#ifndef SORBET_LSP_TASK_H
+#define SORBET_LSP_TASK_H
+
+namespace sorbet::core::lsp {
+// Generic, multi-use task interface.
+// TODO(jvilk): Will use for tasks that preempt IDE slow path typechecking. It's in core because GlobalState will
+// reference it.
+class Task {
```

Comment:
This looks totally reasonable to me, but I don't know off the top of my heads the list of things you're supposed to do when designing a base class in C++, and I vaguely recall some list of rules that included advice like explicitly allowing/disallowing copy constructors?

If you feel good about it, I'm happy to trust that it's either right as-is, or if it's wrong, we can easily fix forward...